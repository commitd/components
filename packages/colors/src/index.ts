export { presetColors } from './preset'
export type { Color } from './preset'
