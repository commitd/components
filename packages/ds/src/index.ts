// component exports
export * from './components'
export * from './utils'

// reexports from ss
export { css, styled } from '@committed/ds-ss'
