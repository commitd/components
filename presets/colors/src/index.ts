export {
  neutralColors,
  preset as presetColors,
  primaryColors,
  semanticColors,
} from './preset'
export type { Color, Prefix, SemanticColor } from './preset'
