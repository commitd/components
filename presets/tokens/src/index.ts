export { preset as presetTokens } from './preset'
