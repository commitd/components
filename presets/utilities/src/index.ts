export { preset as presetUtilities, surfaces } from './preset'
export type { Surface, SurfaceVariants } from './preset'

