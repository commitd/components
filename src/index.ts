export * from './components'
export * from './theme'
export * from './hooks'

// Reexport
export { styled } from '@material-ui/styles'

// export shortcodes for convenience
import * as shortcodes from './components'
export { shortcodes }
