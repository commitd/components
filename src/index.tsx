export * from './components'
export * from './theme'
export * from './styles'
export * from './util'
// Some explicit exports are required to resolve ambiguity
export { ThemeProvider, colors } from './theme'
export { makeStyles, styled, useTheme, withTheme } from './styles'
export type { ThemeProviderProps } from './theme'
export type { ComponentCreator, WithTheme } from './styles'
