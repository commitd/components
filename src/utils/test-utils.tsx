/* eslint-disable @typescript-eslint/no-explicit-any */
/**
 * Import setupTests for your unit tests and you can use `renderLight` and `renderDark` to render elements wrapped in a ThemeProvider.
 * To render without a theme provider use `renderPlain`.
 *
 * N.B. This adds a simple custom id generator so ids match between snapshots.
 *
 */
import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'
import './domrect-polyfill'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import { ThemeProvider } from '..'
import ResizeObserver from 'resize-observer-polyfill'

// Use the polyfill for the ResizeObserver.
// This is used in some components.
global.ResizeObserver = ResizeObserver

const LightTheme: React.FC = ({ children }) => (
  <ThemeProvider choice="light">{children}</ThemeProvider>
)

const DarkTheme: React.FC = ({ children }) => (
  <ThemeProvider choice="dark">{children}</ThemeProvider>
)

export const renderPlain = render

export const renderLight = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: LightTheme, ...options })

export const renderDark = (
  ui: Readonly<React.ReactElement>,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => render(ui, { wrapper: DarkTheme, ...options })

// re-export everything
export * from '@testing-library/react'
export { userEvent }
