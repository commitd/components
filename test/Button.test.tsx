import React from 'react'
import { render } from './test-utils'
import { Box, Button } from '../src'

describe('Button', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(
      <>
        <Box>
          <Button m={1} disabled={false} color="default">
            Default
          </Button>
          <Button m={1} disabled={true} color="default">
            Default
          </Button>
        </Box>
        <Box>
          <Button m={1} disabled={false} color="primary">
            Primary
          </Button>
          <Button m={1} disabled={true} color="primary">
            Primary
          </Button>
        </Box>
        <Box>
          <Button m={1} disabled={false} color="secondary">
            Secondary
          </Button>
          <Button m={1} disabled={true} color="secondary">
            Secondary
          </Button>
        </Box>
        <Box>
          <Button m={1} disabled={false} variant="outlined" color="default">
            Default
          </Button>
          <Button m={1} disabled={true} variant="outlined" color="default">
            Default
          </Button>
        </Box>
        <Box>
          <Button m={1} disabled={false} variant="outlined" color="primary">
            Primary
          </Button>
          <Button m={1} disabled={true} variant="outlined" color="primary">
            Primary
          </Button>
        </Box>
        <Box>
          <Button m={1} disabled={false} variant="outlined" color="secondary">
            Secondary
          </Button>
          <Button m={1} disabled={true} variant="outlined" color="secondary">
            Secondary
          </Button>
        </Box>
        <Box>
          <Button m={1} disabled={false} variant="text" color="default">
            Default
          </Button>
          <Button m={1} disabled={true} variant="text" color="default">
            Default
          </Button>
        </Box>
        <Box>
          <Button m={1} disabled={false} variant="text" color="primary">
            Primary
          </Button>
          <Button m={1} disabled={true} variant="text" color="primary">
            Primary
          </Button>
        </Box>
        <Box>
          <Button m={1} disabled={false} variant="text" color="secondary">
            Secondary
          </Button>
          <Button m={1} disabled={true} variant="text" color="secondary">
            Secondary
          </Button>
        </Box>
      </>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
