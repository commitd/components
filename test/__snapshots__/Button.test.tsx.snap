// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all variants 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Default
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-disabled MuiButtonBase-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Default
      </span>
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-containedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Primary
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-containedPrimary MuiButton-disabled MuiButtonBase-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Primary
      </span>
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-containedSecondary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Secondary
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-containedSecondary MuiButton-disabled MuiButtonBase-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Secondary
      </span>
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Default
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-disabled MuiButtonBase-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Default
      </span>
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-outlinedPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Primary
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-outlinedPrimary MuiButton-disabled MuiButtonBase-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Primary
      </span>
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-outlinedSecondary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Secondary
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-outlinedSecondary MuiButton-disabled MuiButtonBase-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Secondary
      </span>
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Default
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-disabled MuiButtonBase-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Default
      </span>
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-textPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Primary
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-textPrimary MuiButton-disabled MuiButtonBase-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Primary
      </span>
    </button>
  </div>
  <div
    class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-textSecondary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Secondary
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root MuiButton-textSecondary MuiButton-disabled MuiButtonBase-disabled"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Secondary
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`renders dark without crashing 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`renders light without crashing 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
